### PR TITLE
Fix issues identified by go vet 1.21

### PR DESCRIPTION
**Pull request by pmatseykanets**
## Issue: 

```shell
go version
go version go1.21.4 darwin/amd64

go vet ./...
# github.com/rancher/rancher/pkg/controllers/dashboard/chart_test
pkg/controllers/dashboard/chart/chart_test.go:179:38: loop variable tt captured by func literal
# github.com/rancher/rancher/pkg/features
vet: pkg/features/feature_test.go:36:15: undefined: IsDefFalse
```
 
### Automated Testing

* Test types added/modified:
    * Unit
    * Integration (Go Framework)
    * Integration (v2prov Framework)
    * Validation (Go Framework)
    * Other - Explain: _EXPLAIN_
    * None
    * _REMOVE NOT APPLICABLE BULLET POINTS ABOVE_
* If "None" - Reason: _EXPLAIN THE REASON_
  
* If "None" - GH Issue/PR: _LINK TO GH ISSUE/PR TO ADD TESTS_

Summary: _TODO_

### Regressions Considerations

_TODO_

Existing / newly added automated tests that provide evidence there are no regressions:
* _TODO_

### CODE REVIEW

File: pkg/controllers/dashboard/chart/chart_test.go [hunk 1]
```diff
@@ -176,7 +176,7 @@ func TestGetChartValue(t *testing.T) {
 			configCache := fake.NewMockCacheInterface[*v1.ConfigMap](ctrl)
 			test.setup(configCache)
 			getter := chart.RancherConfigGetter{configCache}
-			got, err := getter.GetChartValues(tt.chartName)
```
Comment (tomleb) [inline pkg/controllers/dashboard/chart/chart_test.go:179]: Unless you see any reason not to, could we either have:

```go
for _, tt := range tests {
    tt := tt
```

or

```go
for _, test := range tests {
    test := test
```
?

That way we don't risk using the "wrong" one in the future.

Also, unrelated, but can't wait for https://tip.golang.org/wiki/LoopvarExperiment to be released so we no longer have to use this trick.

**Review by tomleb — CHANGES_REQUESTED** [review]
One small change I think we could do otherwise seems good.

File: pkg/controllers/dashboard/chart/chart_test.go (hunk 1 above)
Comment (pmatseykanets) [reply pkg/controllers/dashboard/chart/chart_test.go:179]: This was author's preference. Although I personally think that either of suggested variants would be better it's not this PR's intent to change that preference. And it's not the only place where this naming was used.

Comment (pmatseykanets) [reply pkg/controllers/dashboard/chart/chart_test.go:179]: > Also, unrelated, but can't wait for https://tip.golang.org/wiki/LoopvarExperiment to be released so we no longer have to use this trick.
Agreed, and we can already use it with Go 1.21 (not by default though).

**Review by tomleb — DISMISSED** [review]

**Review by KevinJoiner — DISMISSED** [review]

**Review by crobby — DISMISSED** [review]

**Review by crobby — APPROVED** [review]

**Review by tomleb — APPROVED** [review]

**Pull request merged**